src/app/modules/verification/__tests__/verification.controller.new.spec.ts

```diff
@@ -0,0 +1,121 @@
+import { ObjectId } from 'bson'
+import { Response } from 'express'
+import { StatusCodes } from 'http-status-codes'
+import mongoose from 'mongoose'
+import { errAsync, okAsync } from 'neverthrow'
+import { mocked } from 'ts-jest/utils'
+
+import { IVerificationSchema } from 'src/types'
+
+import dbHandler from 'tests/unit/backend/helpers/jest-db'
+
+import expressHandler from '../../../../../tests/unit/backend/helpers/jest-express'
+import { DatabaseError } from '../../core/core.errors'
+import * as VerificationController from '../verification.controller'
+import { TransactionNotFoundError } from '../verification.errors'
+import { VerificationFactory } from '../verification.factory'
+import getVerificationModel from '../verification.model'
+
+const VerificationModel = getVerificationModel(mongoose)
+
+jest.mock('../verification.factory')
+const MockVerificationFactory = mocked(VerificationFactory, true)
+
+// eslint-disable-next-line @typescript-eslint/no-empty-function
+const noop = () => {}
+const MOCK_FORM_ID = new ObjectId().toHexString()
+const MOCK_TRANSACTION_ID = new ObjectId().toHexString()
+
+describe('Verification controller', () => {
+  let mockTransaction: IVerificationSchema
+  let mockRes: Response
+
+  beforeAll(async () => {
+    await dbHandler.connect()
+    mockTransaction = await VerificationModel.create({
+      _id: MOCK_TRANSACTION_ID,
+      formId: MOCK_FORM_ID,
+      expireAt: new Date(),
+      fields: [],
+    })
+  })
+
+  beforeEach(() => {
+    mockRes = expressHandler.mockResponse()
+  })
+
+  afterEach(() => jest.resetAllMocks())
+
+  afterAll(async () => {
+    // mockTransaction is reused throughout the tests
+    await dbHandler.clearDatabase()
```

Comment:
in the previous PR, `clearDatabase` is done in the `afterEach` block but here, it is done `afterAll` - would this create bugs where previous transactions interfere with tests and create flaky tests?

Reply:
as commented